pkg/test/step.go

```diff
@@ -402,8 +413,11 @@ func (s *Step) Run(namespace string) []error {
 		return testErrors
 	}
 
-	for i := 0; i < s.GetTimeout(); i++ {
-		testErrors = s.Check(namespace)
+	timeoutF := float64(s.GetTimeout())
```

Comment:
this needed some attention.. is there a reason to change it on this PR?  could you open this separately please.

Reply:
Let me explain why this is I consider this necessary in this PR. As commands for assertions most likely will be far less than instant, the previous implementations won't respect the timeout at all. Let's assume the command takes 10 seconds to execute, and the timeout is 30 seconds, this means that the loop would be running for 330 seconds (10*30 + 1*30) as each loops executes it (10 secs) and just waits 1 second.

Reply:
yeah... I've been frustrated with the "meaning" of timeout and our inconsistent use of timeout for awhile.  I imagined it needed to be address.  I just thought we would have some discussion around it.. etc.    The problem with it in this PR is it is an important component lost in the noise of another important component.  It establishes an opinion which should be debated a bit.

Things to figure out:
1. what does timeout apply to... it could be the testsuite timeout, it could be a test, or a test step.  It could be each independent element in the teststep...
2. The timeout could be separate for the apply phase and assert phase or be inclusive
3. When a timeout change is made it is common to have it made for the assert phase (with knowledge that it takes extra time for the apply to be realized in the assert).  Does that affect assert only.  Is it all asserts collectively.

I also imagine that AssertCommands, might make sense to have a timeout themselves (which is currently missing).  There is the length of time allow for a successful invocation, there is the number of times you are willing to make the invocation (defined here.. as the number of times it can be called within an assert timeout.).

Timeouts can be very important for eventually consistent tests... I believe we have a number of timeouts conflated conceptually together and we need to clean up what a timeout means at different levels of usage.

I will look past it here.  Your position makes sense.   I'm really excited with contributors add value to through PRs as you have and look to provide guidance... but also want to make sure the burden isn't high.  Thanks for the contribution!

Reply:
I think I get your point,  but I would say the modification of the code, through maybe opinionated, has a very small scope and it's not taking a big step in any direction. In other words, what ever discussion and design choice made later does not get impacted by applying this PR or not, the diff of this code and the previous one is minimal. Of course that is my opinion from outside,  what ever you guys decide I'll get it implemented.